src/metabase/query_processor/middleware/visualization_settings.clj

```diff
@@ -0,0 +1,72 @@
+(ns metabase.query-processor.middleware.visualization-settings
+  (:require [medley.core :as m]
+            [metabase.models.card :refer [Card]]
+            [metabase.query-processor.store :as qp.store]
+            [metabase.shared.models.visualization-settings :as mb.viz]
+            [toucan.db :as db]))
+
+(defn- normalize-field-settings
+  [id settings]
+  (mb.viz/db->norm-column-settings {(mb.viz/norm->db-column-ref {::mb.viz/field-id id}) settings}))
+
+(defn- update-card-viz-settings
+  "For each field, fetch its settings from the QP store, convert the settings into the normalized form
+  for visualization settings, and then merge in the card-level column settings."
+  [fields column-viz-settings]
+  (map (fn [[_ id _]]
+         (->> id
+              qp.store/field
+              :settings
+              (#(normalize-field-settings id %))
```

Comment:
You could just do this one as `(normalize-field-settings id)`. `->>` threads thru the last arg. So it will do the same thing

Reply:
This would maybe be a little clearer with a `for` loop as well